src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -0,0 +1,269 @@
+  utils::SharedPtr<NiceMock<resumption_test::MockResumptionData> >
+      mock_storage_;
+  NiceMock<con_test::MockConnectionHandler> mock_connection_handler_;
+  NiceMock<protocol_handler_test::MockSessionObserver> mock_session_observer_;
+  //  NiceMock<hmi_message_handler_test::MockHMIMessageHandler>
+  //      mock_hmi_msg_handler_;
+  NiceMock<MockApplicationManagerSettings> mock_application_manager_settings_;
+  application_manager_test::MockApplicationManager app_mngr_;
+  std::auto_ptr<am::ApplicationManagerImpl> app_manager_impl_;
+  application_manager::MockMessageHelper* mock_message_helper_;
+  application_manager::MessageHelper* message_helper_;
+};
+
+struct AppIdExtractor {
+  AppIdExtractor() {
+    hmi_app_id_ = 0u;
+    is_called_ = false;
+  }
+
+  bool operator()(
+      const hmi_message_handler::MessageSharedPointer& message) const {
+    using namespace application_manager::strings;
+    sync_primitives::AutoLock auto_lock(state_lock_);
+    bool retVal = false;
+    Json::Reader reader;
+    Json::Value json_value;
+    const std::string str = message->json_message();
+    if (reader.parse(str, json_value, false)) {
+      hmi_app_id_ = json_value[params][applications][0][app_id].asUInt();
+      retVal = true;
+    }
+    is_called_ = true;
+    state_condition_.Broadcast();
+    return retVal;
+  }
+};
+
+TEST_F(ApplicationManagerImplTest, ProcessQueryApp_ExpectSuccess) {
+  using namespace NsSmartDeviceLink::NsSmartObjects;
+  SmartObject app_data;
+  const uint32_t connection_key = 65537u;
+  const uint32_t app_id = 0u;
+  app_data[am::json::name] = "application_manager_test";
+  app_data[am::json::appId] = app_id;
+  app_data[am::json::android] = "bucket";
+  app_data[am::json::android][am::json::packageName] = "com.android.test";
+  smart_objects::SmartObject sm_object(SmartType_Map);
+  sm_object[am::json::response][0] = app_data;
+
+  //  EXPECT_CALL(mock_hmi_msg_handler_,
+  //  SendMessageToHMI(Truly(AppIdExtractor())))
+  //      .Times(AtLeast(1u));
+
+  smart_objects::SmartObjectSPtr module_info =
+      utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  smart_objects::SmartObject& object = *module_info;
+  object[strings::params][strings::message_type] = static_cast<int>(kRequest);
+  object[strings::params][strings::function_id] = static_cast<int>(1);
+  object[strings::params][strings::correlation_id] =
+      app_manager_impl_
+          ->GetNextHMICorrelationID();  // GetNextHMICorrelationID();
+  object[strings::msg_params] =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  ON_CALL(*mock_message_helper_, CreateModuleInfoSO(_, _))
+      .WillByDefault(Return(module_info));
+
+  ON_CALL(*mock_message_helper_, CreateNegativeResponse(_, _, _, _))
+      .WillByDefault(Return(module_info));
+
+  app_manager_impl_->ProcessQueryApp(sm_object, connection_key);
+  // Can't test that functionality ofr the moment.
```

Comment:
@pold500 typo and commented code. Such comment can live in code from year to year and nobody will newer fix it.